model/riscv_insts_zfa.sail

```diff
@@ -685,54 +685,57 @@ function clause execute(RISCV_FLTQ_D(rs2, rs1, rd)) = {
 
 /* FCVTMOD.W.D */
 
+/* Calculate the low 32 bits of the integer value from the binary64
+ * floating-point number and return them together with an indicator
+ * whether the NV (overflow) or NX (inexact) exceptions should be raised.
+ */
+val      fcvtmod_helper_inner : forall 'm, 'm >= 0. (bits(1), int('m), bits(53)) -> (bits(5), bits(32))
+function fcvtmod_helper_inner(sign, true_exp, true_mant) = {
+    /* perform the exponentation on the fixed-point
+       mantissa and extract the integer part */
+    let fixedpoint = to_bits(84, unsigned(true_mant)) << true_exp;
+    let integer    = unsigned(fixedpoint[83..52]);
+    let fractional = to_bits(52, unsigned(fixedpoint[51..0]));
+
+    /* apply the sign bit */
+    let result = if (sign == 0b1) then negate(integer)
+                                  else integer;
+
+    /* raise FP exception flags, honoring the precedence of nV > nX */
+    let flags : bits(5) = if (true_exp > 31)         then nvFlag()
+                     else if (fractional != zeros()) then nxFlag()
+                     else                            zeros();
+
+    (flags, to_bits(32, result))
+}
+
 /*
  * Implement float64 to int32 conversion without saturation, the
  * result is supplied modulo 2^32.
- * Rounding mode for FCVTMOD.W.D is always RTZ(Round toward zero).
  */
-val      fcvtmod_helper : forall 'm, 'm == 64. (bits('m)) -> (bits(5), xlenbits)
+val      fcvtmod_helper : forall 'm, 'm == 64. (bits('m)) -> (bits(5), bits(32))
```

Comment:
Why not just bits(64) instead of `bits('m)` where `'m == 64`?

Reply:
This is a double-precision FP bitpattern (i.e. binary64) and unrelated to xlenbits.
It could be `bits(64) -> (bits(5), bits(32))`, though.